### ISSUE TITLE
False positive: Tuple claimed unsubscriptable when using NewType

### ISSUE THREAD

**Issue by Kwasniok** [PyCQA/pylint#4944]
### Bug description

```python
# pylint: disable=C0114,W0104

from typing import NewType

MyTuple = NewType("T", tuple[float, float])
x = MyTuple((1.1, 2.2))
x[0]
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:7:0: E1136: Value 'x' is unsubscriptable (unsubscriptable-object)


------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.75/10, +0.00)
```

### Expected behavior

```

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.2
Python 3.9.6 (default, Jun 30 2021, 10:22:16) 
[GCC 11.1.0]
```

### OS / Environment

```
cat /proc/version
Linux version 5.13.10-arch1-1 (linux@archlinux) (gcc (GCC) 11.1.0, GNU ld (GNU Binutils) 2.36.1) #1 SMP PREEMPT Thu, 12 Aug 2021 21:59:14 +0000
```

### Additional dependencies

# None

**Comment by ggirdler** [comment]
Hi, I'd like to take a look at this issue

**Comment by Pierre-Sassoulas** [comment]
Neat, I assigned you to the issue, to not hesitate to ask for help here or in a draft merge request :)

**Comment by Kwasniok** [comment]
Tank you guys! I am really happy someone is taking a look at this : )

**Comment by cdce8p** [comment]
@ggirdler Just a heads up. This is a more advanced issue. It will require a custom inference tip in `astroid`. If that's your first time working with the pylint and astroid internals, this might not be the right one for you.
